devlib/utils/types.py

```diff
@@ -68,6 +68,15 @@ def numeric(value):
     """
     if isinstance(value, int):
```

Comment:
No, because if we get `1.0`, we want to return an `1` -- this function is intended to prefer ints, so if a float value can be represented as an integer without loss of precision, it will return the corresponding int.